### PR TITLE
Fix lightgbm model input conversion when input is list of lists

### LINKED ISSUE #3221: KServe lgbserver runtime error input for v1 endpoint
/kind bug

**What steps did you take and what happened:**
lgbserver used to support both of following inputs formats prior to 0.11 release.

```python
request = {'inputs' : [{'sepal_width_(cm)': {0: 3.5}, 
'petal_length_(cm)': {0: 1.4}, 
'petal_width_(cm)': {0: 0.2},
'sepal_length_(cm)': {0: 5.1} }]}
```

```python
request2 = {'inputs': [
[{'sepal_width_(cm)': 3.5}, 
 {'petal_length_(cm)': 1.4}, 
 {'petal_width_(cm)': 0.2}, 
 {'sepal_length_(cm)': 5.1}]
 ] }
```

KServe only documented the first input format https://kserve.github.io/website/0.11/modelserving/v1beta1/lightgbm

and in 0.11 the second input format stop working with following error

```bash
2023-11-03 09:06:02.099 32367 kserve ERROR [inference_error_handler():89] Exception:
Traceback (most recent call last):
  File "/Users/dsun20/kserve/python/lgbserver/lgbserver/model.py", line 62, in predict
    result = self._booster.predict(instances)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/lightgbm/basic.py", line 4220, in predict
    return predictor.predict(
           ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/lightgbm/basic.py", line 1047, in predict
    preds, nrow = self.__pred_for_np2d(
                  ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/lightgbm/basic.py", line 1187, in __pred_for_np2d
    return self.__inner_predict_np2d(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/lightgbm/basic.py", line 1127, in __inner_predict_np2d
    data = np.array(mat.reshape(mat.size), dtype=np.float32)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: float() argument must be a string or a real number, not 'dict'
```

**What did you expect to happen:**

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3221

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/predictor/test_lightgbm.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_lightgbm_kserve():
     kserve_client.create(isvc)
     kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
 
-    res = predict(service_name, "./data/iris_input_v3.json")
+    res = predict(service_name, "./data/iris_input_v4.json")
```
Comment (yuzisun) [inline test/e2e/predictor/test_lightgbm.py]: Can we test both inputs?

Comment (sivanantha321) [reply test/e2e/predictor/test_lightgbm.py]: There is already a test for another format

**Comment by yuzisun** [comment]
Thanks @sivanantha321 !!


/lgtm
/approve

**Comment by yuzisun** [comment]
> Thanks @sivanantha321 !!
> 
> /lgtm /approve

Can you also help cherry pick to the 0.11 branch?

**Pull request merged**